### PR TITLE
Tsd UI rajat new

### PR DESCRIPTION
**Pull request by rajatkhade**
suggest any changes in navbar

### CODE REVIEW

**Comment by subhamtsd** [comment]
Hi @rajatkhade 

Everytime you commit code is using your personal account. Pls change your code config

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,18849 @@
+{
```
Comment (subhamtsd) [inline package-lock.json:1]: Hi @rajatkhade 

why you have added package.lock,json

File: package.json [hunk 2]
```diff
@@ -20,7 +20,6 @@
     "jsoneditor": "^9.1.9",
     "jsoneditor-react": "^3.1.0",
```
Comment (subhamtsd) [inline package.json:21]: hi @rajatkhade 

Connect with @nihal772 

to make same package.json with him I have reviewed his package.json

File: packages/demo/components/DetailComponent/ShowEntity.tsx [hunk 3]
```diff
@@ -18,13 +18,20 @@ import {
   Pressable,
 } from "react-native";
 
+import { Feather } from '@expo/vector-icons';
```
Comment (subhamtsd) [inline packages/demo/components/DetailComponent/ShowEntity.tsx]: Hi @rajatkhade 

Donot import like this use normal way import all package all at one line

Comment (subhamtsd) [reply packages/demo/components/DetailComponent/ShowEntity.tsx]: Hi @nihal772  @rajatkhade 

Check with this code I guess you need to move this code to TSD folder

File: packages/demo/examples/TSDigisolPlatform/components/HierarchyComponent/index.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      }
+    ).then((res) => {
+      return res.json()
+    })
+      .then((res) => {
+        console.log("required data in display ", res);
+        setDisplayDataModal(res)
+      });
+  };
+  const fetchCatData = async () => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/HierarchyComponent/index.tsx:57]: hi @rajatkhade 

Move this code to helper component with some meaningfull function name

File: packages/demo/examples/TSDigisolPlatform/components/HierarchyComponent/index.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  };
+  useEffect(() => {
+    fetchData();
+  }, [])
+
+  return (
+    <View>
+      <View>
+        {displayDataModal.response.map((item, index) => <TouchableOpacity 
+          onPress={()=>
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/HierarchyComponent/index.tsx:86]: hi @rajatkhade 

This handler should be moved to events and not here

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/DataModal.tsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                                            }}>{displayModule.modules[selectedIndex].moduleDisplayName}</Text>: <Text style={{
+                                                marginTop: -15,
+                                                fontSize: 24,
+                                                color: "white",
+                                                fontWeight: "bold",
+                                                padding: 10,
+                                            }}>{displayModule.functionName}</Text>}
+                                <TouchableOpacity
+                                    style={{marginLeft:"60%",alignSelf:"flex-end"}}
+                                    onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/DataModal.tsx:83]: Hi @rajatkhade 

You have not added handler to event config

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/SideNavbar.tsx [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+                visible={modalVisible}
+            >
+                <View style={sidenavStyles.centeredView}>
+
+                    <View style={{ justifyContent: "flex-start", height: "100%", }}>
+                        <View style={sidenavStyles.modalView}>
+                            <View style={{ height: "100%" }}>
+                                <View style={{display:"flex",flexDirection:"row"}}>
+                                {tabView ? <TouchableOpacity
+                                    onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/SideNavbar.tsx]: Same here

Move handler to event config

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/SideNavbar.tsx [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+                                            }}>{displayModule.modules[selectedIndex].moduleDisplayName}</Text>: <Text style={{
+                                                marginTop: -15,
+                                                fontSize: 24,
+                                                color: "white",
+                                                fontWeight: "bold",
+                                                padding: 10,
+                                            }}>{displayModule.functionName}</Text>}
+                                <TouchableOpacity
+                                    style={{alignItems:"flex-end"}}
+                                    onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/SideNavbar.tsx]: Same here as above one

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/SideNavbar.tsx [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+                                            fontSize: 14,
+                                            color: "white",
+                                            padding: 10,
+                                        }}>
+                                            {item.tabDisplayName}
+                                        </Text>
+                                    </TouchableOpacity>)}
+                                </View> : <View>{displayModule.modules.map((item, key) => (
+                                    <TouchableOpacity
+                                        onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/SideNavbar.tsx]: Same here as above

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+    // console.log("data of BusinessFunctions :", listDataSource);
+    // console.log("data of modules :", modules);
+
+    return (
+        <View style={{ height: "100vh", display: "flex", flexDirection: "row", position: "fixed",marginTop:-35 }}>
+            <View style={{ height: "100%", width: "90px", backgroundColor: "#373330", alignItems: "center" }}>
+                <View style={{ backgroundColor: activeFunction==="foundation"?"#4a4542":"#373330",width:"100%",paddingVertical:"20%" ,marginTop: "10vh", alignItems: "center" }}>
+                    {/* hard coded because not sure how to get the icons */}
+                    <TouchableOpacity
+                        onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx:76]: Move the same to event and not here 
@rajatkhade

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+                        <Text
+                            style={{color:"#aaa6a0",fontSize:10,fontWeight:"bold"}}
+                        >
+                            FOUNDATION
+                        </Text>
+                    </TouchableOpacity>
+                </View>
+                <View style={{ backgroundColor: activeFunction==="sales"?"#4a4542":"#373330",width:"100%",paddingVertical:"30%", alignItems: "center" }}>
+                    <TouchableOpacity
+                        onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx:98]: Hi @rajatkhade 

Same here

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+                            size={40} 
+                            color="#aaa6a0"
+                            style={{marginBottom: 5, alignSelf: "center" }}    
+                        />
+                        <Text style={{color:"#aaa6a0",fontSize:10,fontWeight:"bold"}}>SALES</Text>
+                    </TouchableOpacity>
+                </View>
+                <View style={{ backgroundColor: activeFunction==="postsales"?"#4a4542":"#373330",width:"100%",paddingVertical:"30%", alignItems: "center" }}>
+                    <TouchableOpacity
+                        onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx:116]: same here as above

File: packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+                            size={24} 
+                            color="#aaa6a0"
+                            style={{marginBottom: 15, alignSelf: "center" }}   
+                        />
+                        <Text style={{color:"#aaa6a0",fontSize:10,fontWeight:"bold"}}>POST SALES</Text>
+                    </TouchableOpacity>
+                </View>
+                <View style={{ backgroundColor: activeFunction==="configurator"?"#4a4542":"#373330",width:"100%",paddingVertical:"30%", alignItems: "center" }}>
+                    <TouchableOpacity
+                        onPress={() => {
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/components/NewNavbar/index.tsx:134]: same as above

File: packages/demo/examples/TSDigisolPlatform/configs/layouts/dashboardLayout.js [hunk 14]
```diff
@@ -43,8 +43,8 @@ import { EditOrderLineAddressDetailComponent } from "../../components/domainSpec
 // ******************** TEST OF AddEditEntity *************************
 import { AddEditEntity } from "../../../../components/AddEditEntity";
 
-// ******************** POC on NAV BAR *****************************
-import { ToggleNavigation } from "../../../../components/ToggleNavigation";
+// ******************** POC on NEWNAV BAR *****************************
+import { NewNavbar } from "../../components/NewNavbar/index";
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/configs/layouts/dashboardLayout.js:47]: Hi @rajatkhade 

Can you let me know where does you have appConfig 
for our repo We are using appConfig written in layout.js file 
Which file you are using ?

File: packages/demo/examples/TSDigisolPlatform/configs/routes/routesConfig.js [hunk 15]
```diff
@@ -299,7 +299,7 @@ routes.orderDetail = {
     },
   },
 };
-
+//demo checking
```
Comment (subhamtsd) [inline packages/demo/examples/TSDigisolPlatform/configs/routes/routesConfig.js]: Hi @rajatkhade 

Connect with @nihal772 
for this file I have reviews his file on this

File: packages/demo/examples/TSDigisolPlatform/components/HierarchyComponent/index.tsx (hunk 4 above)
Comment (rajatkhade) [reply packages/demo/examples/TSDigisolPlatform/components/HierarchyComponent/index.tsx:57]: This component is still under development

File: packages/demo/examples/TSDigisolPlatform/configs/layouts/dashboardLayout.js (hunk 14 above)
Comment (rajatkhade) [reply packages/demo/examples/TSDigisolPlatform/configs/layouts/dashboardLayout.js:47]: I'm also using the same

**Pull request merged**